opendatafabric/src/serde/serdes.rs

```diff
@@ -77,6 +120,11 @@ pub enum Error {
         manifest_version: i32,
         supported_version: i32,
     },
+    #[error("Obsolete version: manifest has version {manifest_version} while minimum supported version is {minimum_supported_version}")]
+    ObsoleteVersion {
```

Comment:
Another approach could be to have one `UnsupportedVersionError` struct with `supported_version_range` and then define `impl Display` that shows one message if actual version is below the supported range (obsolete) and another message if it's above the range.

Having one error instead of two might simplify things, but leaving it up to you.

Reply:
Yes, interesting. I was able to push the version range as enum instead, and only expanded to i32 while formatting.